File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def count_prime(n):
    if n < 2:
        return 0
    
    result = [True] * (n+1)
    result[0] = result[1] = False
    
    for i in range(2, int(n ** 0.5) + 1):
        if result[i]:
            for j in range(i*i, n+1, i):
                result[j] = False
    return sum(result)

def solution(n):
    return count_prime(n)

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
function countPrimeNum(n){
    if(n < 2) return 0;
    
    let result = Array(n+1).fill(true);
    result[0] = false, result[1] = false;
    
    for(let i = 2 ; i < Math.floor(n ** 0.5) + 1; i++){
        if(result[i]){
            for(let j = i*i ; j <= n ; j += i){
                result[j] = false;
            }
        }
    }
    
    return result.reduce((acc, cur) => acc + Number(cur), 0);
    
}
function solution(n) {
    return countPrimeNum(n);
}